proto/sf/ethereum/type/v2/type.proto

```diff
@@ -48,6 +48,26 @@ import "google/protobuf/timestamp.proto";
 // is not reliable and should not be used to order the element against other
 // elements in the block as those element might have 0 as the ordinal. Only
 // successful calls have a reliable `ordinal` field.
+//
+// ## Failed Transaction State Changes
+//
+// An important edge case to understand when processing Firehose blocks is that
+// certain state changes persist even when a transaction fails. This behavior
+// follows Ethereum's execution model where some operations are irreversible
+// once they occur, regardless of transaction outcome.
+//
+// The following state changes are typically preserved even for failed transactions:
+// - Gas consumption (REASON_GAS_BUY, REASON_INTRINSIC_GAS): Gas is always consumed
+//   when a transaction is processed, even if execution fails
+// - Gas refunds (REASON_GAS_REFUND): Gas refunds are processed regardless of transaction outcome
+// - Nonce increments: The sender's nonce is incremented to prevent replay attacks
+// - Transaction fees (REASON_REWARD_TRANSACTION_FEE): Miners receive fees regardless
+//   of transaction success to compensate for computational work
```

Comment:
Ok, when reading this, it's unclear that for those with REASON, we talk about BalanceChange, let's contexify a bit more each line of element because I was confusing it with GasChange.